server/src/test/java/org/apache/druid/server/coordinator/simulate/HistoricalCloningTest.java

```diff
@@ -0,0 +1,292 @@
+                                                         .withUnmanagedNodes(Set.of(historicalT12.getHost()))
+                                                         .withSmartSegmentLoading(false)
+                                                         .build()
+                             )
+                             .withImmediateSegmentLoading(true)
+                             .build();
+
+    Segments.WIKI_10X100D.forEach(segment -> historicalT11.addDataSegment(segment));
+    startSimulation(sim);
+
+    // Run 1: All segments are loaded on the source historical
+    runCoordinatorCycle();
+    Assert.assertEquals(1000, historicalT11.getTotalSegments());
+    Assert.assertEquals(0, historicalT12.getTotalSegments());
+
+    // Clone server now added.
+    addServer(historicalT12);
+
+    // Run 2: Assigns all segments to the cloned historical
+    runCoordinatorCycle();
+
+    Assert.assertEquals(1000, historicalT11.getTotalSegments());
+    Assert.assertEquals(1000, historicalT12.getTotalSegments());
+
+    verifyValue(
+        Stats.Segments.CLONE_LOAD.getMetricName(),
+        Map.of("server", historicalT12.getName()),
+        1000L
+    );
+
+    verifyValue(
+        Metric.SUCCESS_ACTIONS,
+        Map.of("server", historicalT12.getName(), "description", "LOAD: NORMAL"),
+        1000L
+    );
+  }
+
+  @Test
+  public void testCloningHistoricalWithReplicationLimit()
+  {
+    final CoordinatorSimulation sim =
+        CoordinatorSimulation.builder()
+                             .withSegments(Segments.WIKI_10X1D)
+                             .withServers(historicalT11, historicalT12, historicalT13)
+                             .withRules(datasource, Load.on(Tier.T1, 2).forever())
+                             .withImmediateSegmentLoading(true)
+                             .withDynamicConfig(
+                                 CoordinatorDynamicConfig.builder()
+                                                         .withCloneServers(Map.of(historicalT11.getHost(), historicalT12.getHost()))
+                                                         .withUnmanagedNodes(Set.of(historicalT12.getHost()))
+                                                         .withSmartSegmentLoading(false)
+                                                         .withReplicationThrottleLimit(2)
+                                                         .withMaxSegmentsToMove(0)
+                                                         .build()
+                             )
+                             .withImmediateSegmentLoading(true)
+                             .build();
+    Segments.WIKI_10X1D.forEach(historicalT13::addDataSegment);
+    startSimulation(sim);
+
+    // Check that only replication count segments are loaded each run and that the cloning server copies it.
+    while (historicalT11.getTotalSegments() < Segments.WIKI_10X1D.size()) {
```

Comment:
Since we are doing things in a loop here, please add a timeout to the test and assert the number of loop iterations in the end.